src/ng/directive/select.js

```diff
@@ -113,36 +131,128 @@ var SelectController =
     return !!optionsMap.get(value);
   };
 
+  var handleMultipleChanges = false;
+  function updateModelAfterOptionChange(renderAfter) {
+    if (self.multiple) {
+      if (!handleMultipleChanges) {
+        handleMultipleChanges = true;
+      } else {
+        $scope.$$postDigest(function() {
+          handleMultipleChanges = false;
+          self.ngModelCtrl.$setViewValue(self.readValue());
+          if (renderAfter) self.ngModelCtrl.$render();
+        });
```

Comment:
This pattern of running something once in the postDigest keeps appearing. I wonder if we should abstract it out into $scope?